src/panels/lovelace/create-element/create-element-base.ts

```diff
@@ -36,11 +38,11 @@ const _createElement = <T extends keyof CreateElementConfigTypes>(
   ) as CreateElementConfigTypes[T]["element"];
   try {
     // @ts-ignore
-    element.setConfig(config);
+    element.setConfig(deepClone(config));
```

Comment:
We shouldn't do this. This hurts our performance because cards are doing the incorrect thing. 

Instead we should use `Object.freeze` and let custom cards crash so they fix their bad behavior. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

Reply:
We should freeze it as soon as we load the config.

Reply:
`Object.freeze()` is shallow, so that would be a recursive process. But I guess it's better to do it once...
![Skärmavbild 2020-02-12 kl  08 45 53](https://user-images.githubusercontent.com/1299821/74313602-4c4ed000-4d74-11ea-8274-412febb728e4.png)

Reply:
Yes, we should deep freeze. Deep freeze example code is on the docs I linked :) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze#What_is_shallow_freeze

Reply:
🤦‍♂